component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/configuration/PropertiesSetup.java

```diff
@@ -45,11 +45,12 @@ public void accept(final Configuration configuration) {
         System.setProperty("geronimo.opentracing.client.filter.request.skip", "true");
         System.setProperty("geronimo.opentracing.filter.skippedTracing.urls", ".*/login$,.*/decrypt/.*");
         System.setProperty("geronimo.opentracing.filter.skippedTracing.matcherType", "regex");
-        // environment dft ordinal 400
-        final String httpPort = System.getenv("TALEND_COMPONENT_SERVER_PORT");
-        if (httpPort != null) {
-            System.setProperty("http", httpPort);
-            configuration.setHttpPort(Integer.parseInt(httpPort));
+        // listening port order: `meecrowave cli --http=x` > `-Dhttp=x` > `-e TALEND_COMPONENT_SERVER_PORT=x`
+        final String port = System.getProperty("http") != null ? System.getProperty("http")
```

Comment:
Does "--http=x" is assigned to the system property "http" somewhere in the application?

Reply:
not in application directly. Usually from the launching environment (re2, ce,etc.)